### DISCUSSION TITLE
ice-phase (and mixed-phase) support roadmap (attributes/dynamics/products/formulae/examples)

### DISCUSSION THREAD

**Discussion by slayoo** [open-atmos/PySDM#1234] [Ideas]
Just creating a venue to exchange updates and ideas on the ice-phase support in PySDM.

**Comment by tluettm** [comment]
This a short roadmap of an extension of PySDM for the ice phase, particular for the description of cirrus clouds. However, general mechanisms governing the ice phase like sedimentation and growth by water vapor diffusion should also work for the mixed-phase regime.


**Homogeneous freezing of solution droplets:**

Usually homogeneous freezing of solution droplets is described as a threshold process. Commonly used parameterizations of homogeneous nucleation describe an entire nucleation event and provide number of nucleated ice crystals per unit volume (see, e.g., Kärcher and Lohmann 2002). Super Particle models containing ice use these bulk rates to initialize a number of Ice Super Particles (see, e.g., Sölch and Kärcher (2010), Brdar and Seifert (2017)). However, it would be more physical correct to describe homogeneous freezing as a stochastic processes. PySDM should be a good basis  for such a description.

1.  Describe growth of solution droplet in environment subsaturated wrt water (but supersaturated wrt ice) by using PySDMs implementation of Köhler equation and existing aqueous chemistry routines
2. Solution droplets freeze stochastically depending on volume, temperature and supersaturation using recent approximations of homogeneous nucleation rate coefficient (e.g., Spichtinger et al., 2023). Test: how many super particles are needed for a convergence of the stochastic process? 
3. Implement general properties of ice in cirrus. Ice crystals shape is approximated as prolate ellipsoids. More complex shapes and shape prediction can be considered later. They are less important for cirrus but relevant for, e.g., the mixed-phase regime  
4. Add new routine as an adaption of the condensation routine for growth by diffusion of water vapor for ice crystals. Use up to date formulations for capacity, kinetic correction and ventilation correction. Problem: what happens when ice crystal (formed by homogeneous nucleation) evaporate? Transform ice crystal into solution droplet of equal mass?
5. Implement vertical displacement by sedimentation. Can use the same routine as for liquid droplets?
6. Aggregation: can be implemented as a Monte-Carlo scheme (based on Shima et al., 2009) similar to coalescence routine (see,e.g., Sölch and Kärcher (2010), Brdar and Seifert (2017)). However, aggregation is not as important for cirrus clouds. Implement later. 


**Deposition nucleation:**

There are a lot of unknowns regarding deposition nucleation. Ice Nucleating Particles (INP) show a large variability regarding their activation depending on aerosol type and chemical composition. Parameterizations of INP active sites and activation schemes even of the same aerosol type vary strongly between studies. Instead of choosing specific aerosol species and activation schemes, we can adapt the approach of Spichtinger and Cziczo (2010) to capture general trends of heterogeneous ice nucleation . Instead of choosing aerosol modes and size distribution, one can construct a distribution of INP as a function of their nucleation threshold. Of course one still has to constrain this distribution with laboratory studies. But this less of a hassle than trying to explicitly model the ice nucleating properties of specific aerosol species.

Usually, as for homogeneous nucleation,  deposition nucleation parameterizations use threshold processes and freeze a fraction of the INP population, which provides a number of nucleated ice crystals per unit volume. However,  we could also try to formulate deposition nucleation stochastically.

**Reply by slayoo** [reply]
For the record, here's a first PR introducing a new mixed-phase example from Abade & Albuquerque 2024: https://github.com/open-atmos/PySDM/pull/1365 (for now, only bulk phase partitioning implemented as in Fig 2c in the paper)

**Reply by slayoo** [reply]
and here's a recap of discussion with @tluettm in Kraków:
- work in progress on deposition: #1389 
- roadmap:
  - [ ] latent heat budget (perhaps refactoring the ambient tendency handling)
  - [x] subsaturated vapour diffusion test
  - [ ] melting as a separate dynamic
  - [ ] refactornig of the new diffusion dynamic to be integrated with current condensation (renamig to vapour diffusion), which will give us: implicit-in-mass solution, counters (WBF, homog/heterog. nucleation, ...), adaptative timestepping
  - [ ] homogeneous freezing handling
  - [ ] tests with ABIFM vs. INAS for the Abade & Alb. example
  - [ ] Abade & Alb. turbilence model
  - [ ] refactor diffusion coordinate to switch from volume to mass (#1390)
  - [ ] GPU support
  - [ ] refactor passing of ambient saturation-related vars not to pass redundant RH and a_w_ice if we only need S_ice in a method
